docs/reference/developer/tooling.rst

```diff
@@ -1,38 +1,60 @@
 Tooling
 =======
 
-This is a list of the tooling (i.e. compilers, static analysis, etc.) that are used in Scipp and their corresponding versions.
+This is a list of first party and third party tools used for developing Scipp.
+
+Third Party
+^^^^^^^^^^^
+
+Compilers
+~~~~~~~~~
+
+All compilers can be installed through conda.
+See ``conda/meta.yaml``.
+
+- GCC [Linux]
+- XCode [macOS]
+- MSVC [Windows]
+
+Static Analysis and Formatters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We use ``tox`` and ``pre-commit`` to do auto formatting and static analysis.
+See in particular ``pre-commit-config.yaml`` for the list of used tools and versions.
 
 Misc
 ~~~~
 
-- CMake >= 3.16
+- CMake >= 3.21
 - Conda
 
 Note: Ubuntu users can use the `Kitware Repo <https://apt.kitware.com/>`_ to obtain the latest version.
 
-Compilers
-~~~~~~~~~
+First Party
+^^^^^^^^^^^
 
-- GCC >= 9
-- XCode >= 12.4
-- MSVC >= 16 (2019)
+There are several development tools in the ``tools`` directory at the top level of the Scipp repository.
+See the readmes in the folder.
 
-Static Analysis and Formatters
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Stubgen
+~~~~~~~
+
+A number of classes are defined in C++ and exposed to Python via Pybind11.
+These classes are invisible to static analysis tools, in particular type checkers.
+In addition, some classes have many dynamically bound methods which are equally invisible to type checkers.
+
+So a stubfile is provided which lists these classes and all their methods.
+This file is automatically generated by the ``stubgen`` package located in ``tools/stubgen/``.
+It can be used by calling
 
-- ``clang-format`` 10.0
-- ``cmake-format`` 0.6.9
-- ``flake8``
-- ``yapf`` 0.32.0
+.. code-block:: bash
 
-``clang-format`` may be installed via the LLVM repositories.
+    cd tools
+    python -m stubgen
 
-``cmake-format``, ``flake8`` and ``yapf`` may be installed via ``pip``:
+A tox environment is also provided: ``tox -e stubgen``.
 
-.. code-block::
+Both ways of using ``stubgen`` require Scipp to be installed (potentially in editable mode) in the current environment.
+See ``tools/stubgen/README.md`` for details.
 
-  pip install \
-    cmake-format==0.6.9 \
-    flake8 \
-    yapf==0.32.0
+Remember to call the stubgen whenever you make a change to the interface of those core classes!
```

Comment:
Maybe this belongs in the "Getting Started" page then?

Can we detect changes in the `lib/python` folder and give a warning/error if the stubs have not been updated?

Reply:
It also needs to be updated when dynamically bound, i.e. defined in `src/scipp`, methods are added/changed.